web-api-shared/src/elastic.rs

```diff
@@ -304,16 +440,39 @@ impl Client {
         })
     }
 
-    pub async fn query_with_bytes<B, T>(
+    pub async fn query_with_bytes<T>(
+        &self,
+        method: Method,
+        url: Url,
+        post_data: Option<(HeaderMap<HeaderValue>, Bytes)>,
+    ) -> Result<T, Error>
+    where
+        T: DeserializeOwned,
+    {
+        self.retry(
+            |err| matches!(err, Error::Transport(_)),
+            || async {
+                let method = method.clone();
+                let url = url.clone();
+                let post_data = post_data.clone();
+                self.query_with_bytes_without_retrying(method, url, post_data)
+                    .await
+            },
+        )
+        .await
+    }
+
+    async fn query_with_bytes_without_retrying<B, T>(
         &self,
         method: Method,
         url: Url,
         post_data: Option<(HeaderMap<HeaderValue>, B)>,
-    ) -> Result<Option<T>, Error>
```

Comment:
It turns out this was very error prone, we did had some subtle bugs where a 404 might get ignored in some places where it should yield an error.

So I replaced it with `T` and a extension trait allowing converting a `Result<T, Error>` into a `Result<Option<T>, Error>` based on weather the error is `Error:ResourceNotFound`.